diff -Nr -U 5 samba_orig/source/configure samba_hurd/source/configure
--- samba_orig/source/configure	2004-07-20 18:33:26.000000000 +0200
+++ samba_hurd/source/configure	2004-08-24 14:15:59.000000000 +0200
@@ -18354,10 +18354,11 @@
 else
   cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 #include "confdefs.h"
 
+#define _GNU_SOURCE
 #if defined(HAVE_UNISTD_H)
 #include <unistd.h>
 #endif
 #include <stdio.h>
 #include <sys/stat.h>
diff -Nr -U 5 samba_orig/source/include/includes.h samba_hurd/source/include/includes.h
--- samba_orig/source/include/includes.h	2004-07-20 18:28:11.000000000 +0200
+++ samba_hurd/source/include/includes.h	2004-08-24 14:15:50.000000000 +0200
@@ -19,10 +19,13 @@
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
 
+#define _GNU_SOURCE
+#define _FILE_OFFSET_BITS 64
+
 #ifndef NO_CONFIG_H /* for some tests */
 #include "config.h"
 #endif
 
 #include "local.h"
diff -Nr -U 5 samba_orig/source/lib/iconv.c samba_hurd/source/lib/iconv.c
--- samba_orig/source/lib/iconv.c	2004-07-20 18:28:03.000000000 +0200
+++ samba_hurd/source/lib/iconv.c	2004-08-20 23:01:45.000000000 +0200
@@ -75,11 +75,11 @@
 static struct charset_functions *find_charset_functions(const char *name) 
 {
 	struct charset_functions *c = charsets;
 
 	while(c) {
-		if (strcasecmp(name, c->name) == 0) {
+		if (c->name && strcasecmp(name, c->name) == 0) {
 			return c;
 		}
 		c = c->next;
 	}
 
diff -Nr -U 5 samba_orig/source/tests/summary.c samba_hurd/source/tests/summary.c
--- samba_orig/source/tests/summary.c	2004-07-20 18:28:01.000000000 +0200
+++ samba_hurd/source/tests/summary.c	2004-08-20 23:01:23.000000000 +0200
@@ -1,14 +1,9 @@
 #include <stdio.h>
 
 main()
 {
-#if !(defined(HAVE_FCNTL_LOCK) || defined(HAVE_STRUCT_FLOCK64))
-	printf("ERROR: No locking available. Running Samba would be unsafe\n");
-	exit(1);
-#endif
-
 #if !(defined(HAVE_IFACE_IFCONF) || defined(HAVE_IFACE_IFREQ) || defined(HAVE_IFACE_AIX))
 	printf("WARNING: No automated network interface determination\n");
 #endif
 
 #if !(defined(USE_SETEUID) || defined(USE_SETREUID) || defined(USE_SETRESUID) || defined(USE_SETUIDX))
